float32 x
float32 y
float32 x_next
float32 y_next
int32 id
string letter
int32 missing_letters
